## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 480040`
Commit: `81579cec97b23d53b4baca9f46625d936985e33e`

---
### `x86_64-linux`
<details>
  <summary>:x: 9 packages failed to build:</summary>
  <ul>
    <li>nitrokey-fido2-firmware</li>
    <li>python313Packages.executorch</li>
    <li>python313Packages.executorch.dist</li>
    <li>python313Packages.torchtune</li>
    <li>python313Packages.torchtune.dist</li>
    <li>python314Packages.llama-stack-client</li>
    <li>python314Packages.llama-stack-client.dist</li>
    <li>python314Packages.torchao</li>
    <li>python314Packages.torchao.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 53 packages built:</summary>
  <ul>
    <li>atomic-operator</li>
    <li>atomic-operator.dist</li>
    <li>carburetor</li>
    <li>creds</li>
    <li>creds.dist</li>
    <li>mokuro</li>
    <li>mokuro.dist</li>
    <li>osc-cli</li>
    <li>osc-cli.dist</li>
    <li>owocr</li>
    <li>owocr.dist</li>
    <li>pynitrokey (python313Packages.pynitrokey)</li>
    <li>pynitrokey.dist (python313Packages.pynitrokey.dist)</li>
    <li>python313Packages.attacut</li>
    <li>python313Packages.attacut.dist</li>
    <li>python313Packages.autofaiss</li>
    <li>python313Packages.autofaiss.dist</li>
    <li>python313Packages.deepface</li>
    <li>python313Packages.deepface.dist</li>
    <li>python313Packages.fire</li>
    <li>python313Packages.fire.dist</li>
    <li>python313Packages.llama-stack-client</li>
    <li>python313Packages.llama-stack-client.dist</li>
    <li>python313Packages.manga-ocr</li>
    <li>python313Packages.manga-ocr.dist</li>
    <li>python313Packages.nkdfu</li>
    <li>python313Packages.nkdfu.dist</li>
    <li>python313Packages.paddleocr</li>
    <li>python313Packages.paddleocr.dist</li>
    <li>python313Packages.pdf2docx</li>
    <li>python313Packages.pdf2docx.dist</li>
    <li>python313Packages.ssg</li>
    <li>python313Packages.ssg.dist</li>
    <li>python313Packages.torchao</li>
    <li>python313Packages.torchao.dist</li>
    <li>python314Packages.attacut</li>
    <li>python314Packages.attacut.dist</li>
    <li>python314Packages.autofaiss</li>
    <li>python314Packages.autofaiss.dist</li>
    <li>python314Packages.fire</li>
    <li>python314Packages.fire.dist</li>
    <li>python314Packages.manga-ocr</li>
    <li>python314Packages.manga-ocr.dist</li>
    <li>python314Packages.nkdfu</li>
    <li>python314Packages.nkdfu.dist</li>
    <li>python314Packages.pdf2docx</li>
    <li>python314Packages.pdf2docx.dist</li>
    <li>python314Packages.pynitrokey</li>
    <li>python314Packages.pynitrokey.dist</li>
    <li>python314Packages.ssg</li>
    <li>python314Packages.ssg.dist</li>
    <li>tractor</li>
    <li>tractor.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>nitrokey-fido2-firmware</summary>
<pre>*** ../../crypto/tiny-AES-c/aes.c
*** ../../crypto/cifra/src/sha512.c
*** ../../crypto/cifra/src/blockwise.c
*** src/main.c
*** src/init.c
src/main.c: In function &#x27;main&#x27;:
src/main.c:55:5: error: too many arguments to function &#x27;device_init&#x27;; expected 0, have 2
   55 |     device_init(argc, argv);
      |     ^~~~~~~~~~~ ~~~~
In file included from src/app.h:11,
                 from ../../fido2/log.h:11,
                 from src/main.c:18:
src/solo.h:4:6: note: declared here
    4 | void device_init();
      |      ^~~~~~~~~~~
make[1]: *** [build/application.mk:67: src/main.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory &#x27;/build/source/targets/stm32l432&#x27;
test -f solo.hex || false
make: *** [Makefile:46: firmware] Error 1</pre>
</details>
<details>
<summary>python314Packages.llama-stack-client</summary>
<pre>[gw11] linux -- Python 3.14.2 /nix/store/pmb8clh66mryha2ijzz3dg969drrkgj7-python3-3.14.2/bin/python3.14
tests/test_models.py:814: in test_discriminated_unions_invalid_data_uses_cache
    m = construct_type(
/nix/store/n69m01z2sqrfb3bi32mlrzx35qr7fvvd-python3.14-llama-stack-client-0.2.23/lib/python3.14/site-packages/llama_stack_client/_models.py:502: in construct_type
    discriminator = _build_discriminated_union_meta(union=type_, meta_annotations=meta)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/nix/store/n69m01z2sqrfb3bi32mlrzx35qr7fvvd-python3.14-llama-stack-client-0.2.23/lib/python3.14/site-packages/llama_stack_client/_models.py:672: in _build_discriminated_union_meta
    cast(CachedDiscriminatorType, union).__discriminator__ = details
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E   AttributeError: &#x27;typing.Union&#x27; object has no attribute &#x27;__discriminator__&#x27; and no __dict__ for setting new attributes
=========================== short test summary info ============================
FAILED tests/test_models.py::test_discriminated_unions_unknown_variant - AttributeError: &#x27;typing.Union&#x27; object has no attribute &#x27;__discriminator__&#x27; ...
FAILED tests/test_models.py::test_discriminated_unions_invalid_data - AttributeError: &#x27;typing.Union&#x27; object has no attribute &#x27;__discriminator__&#x27; ...
FAILED tests/test_models.py::test_discriminated_union_case - AttributeError: &#x27;typing.Union&#x27; object has no attribute &#x27;__discriminator__&#x27; ...
FAILED tests/test_models.py::test_discriminated_unions_invalid_data_nested_unions - AttributeError: &#x27;typing.Union&#x27; object has no attribute &#x27;__discriminator__&#x27; ...
FAILED tests/test_models.py::test_nested_discriminated_union - AttributeError: &#x27;typing.Union&#x27; object has no attribute &#x27;__discriminator__&#x27; ...
FAILED tests/test_models.py::test_discriminated_unions_with_aliases_invalid_data - AttributeError: &#x27;typing.Union&#x27; object has no attribute &#x27;__discriminator__&#x27; ...
FAILED tests/test_models.py::test_discriminated_unions_overlapping_discriminators_invalid_data - AttributeError: &#x27;typing.Union&#x27; object has no attribute &#x27;__discriminator__&#x27; ...
FAILED tests/test_models.py::test_discriminated_unions_invalid_data_uses_cache - AttributeError: &#x27;typing.Union&#x27; object has no attribute &#x27;__discriminator__&#x27; ...
======================== 8 failed, 246 passed in 5.09s =========================</pre>
</details>
</details>

--- a/pkgs/development/python-modules/fire/default.nix
+++ b/pkgs/development/python-modules/fire/default.nix
@@ -9,6 +9,7 @@
   levenshtein,
   pytestCheckHook,
   termcolor,
+  pythonAtLeast,
 }:
 
 buildPythonPackage rec {
@@ -35,6 +36,11 @@ buildPythonPackage rec {
     mock
     levenshtein
     pytestCheckHook
+  ];
+
+  disabledTests = lib.optionals (pythonAtLeast "3.14") [
+    # RuntimeError: There is no current event loop in thread 'MainThread'
+    "testFireAsyncio"
   ];
 
   pythonImportsCheck = [ "fire" ];

